fuel/converters/ilsvrc2010.py

```diff
@@ -0,0 +1,908 @@
+        set images for the corresponding class index.
+    wnid_map : dict
+        A dictionary mapping WordNet IDs to class indices.
+    image_dim : int
+        The width and height of the desired images after resizing and
+        central cropping.
+    num_workers : int
+        The number of worker processes to spawn, in addition to a
+        source and sink process.
+    worker_batch_size : int
+        The number of images each worker should send over the socket
+        to the sink at a time.
+
+    """
+    n_train = sum(train_images_per_class)
+    ventilator = multiprocessing.Process(target=train_set_ventilator,
+                                         args=(train_archive,))
+    ventilator.start()
+    workers = [multiprocessing.Process(target=train_set_worker,
+                                       args=(train_archive,
+                                             patch_archive, wnid_map,
+                                             train_images_per_class,
+                                             image_dim,
+                                             worker_batch_size))
+               for _ in xrange(num_workers)]
+    for worker in workers:
+        worker.start()
+    sink = multiprocessing.Process(target=train_set_sink,
+                                   args=(hdf5_file, n_train,
+                                         train_images_per_class))
+    sink.start()
+    terminate = False
+    try:
+        context = zmq.Context()
+        # Only monitor the ventilator/sink for aliveness. Workers should only
+        # terminate when there's an error.
+        zmq_log_and_monitor(log, context,
+                            processes=[ventilator, sink],
+                            failure_threshold=logging.ERROR)
+    except KeyboardInterrupt:
+        terminate = True
+        log.info('Keyboard interrupt received.')
+    except SubprocessFailure:
+        terminate = True
+        log.info('One or more substituent processes failed.')
+    except Exception:
+        terminate = True
+    finally:
+        log.info('Shutting down workers and ventilator...')
+        for worker in workers:
+            worker.terminate()
+        ventilator.terminate()
+        sink.terminate()
+        log.info('Killed child processes.')
+        if terminate:
+            context.destroy()
+            sys.exit(1)
+
+
+def process_other_set(hdf5_file, archive, patch_archive, groundtruth,
+                      which_set, worker_batch_size, image_dim, offset):
+    """Process and convert either the validation set or the test set.
+
+    Parameters
+    ----------
+    hdf5_file : :class:`h5py.File` instance
+        HDF5 file handle to which to write. Assumes `features`, `targets`
+        and `filenames` already exist and are at least as long as
+        `offset` plus the number of files in `archive`.
+    archive : str or file-like object
+    patch_archive : str or file-like object
```

Comment:
I'll allow myself to be annoying and request more information for the two arguments above.